df_chat/asgi/consumers.py

```diff
@@ -100,6 +100,11 @@ async def unsubscribe_from_all_activities(self, **kwargs):
         """
         Unsubscribe from all rooms
         """
+        # Unauthenticated users are not subscribed.
+        # (This is called on channel close.)
+        if not self.user.is_authenticated:
```

Comment:
Is this the right approach to silently ignore this? Will this not mask some other issues upstream/downstream?

Reply:
This gets called from `disconnct()`. However, for readability, it makes more sense to only check this where it has potential to (needlessly) generate errors. So a least-generic approach - specifically so that any unexpected Exceptions bubble through.